### PR TITLE
Replace native `http/net` with `gin` framework

### PR DESCRIPTION
**Pull request by VitaliLupusor**
Replaced the native package with the `Gin` framework.

Gin provides capabilities of easy mount of middleware onto routers; and request URL manipulation.
Additionally, provides an out of the box method to respond with JSON-like objects to the inbound requests.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 func main() {
-	http.HandleFunc("/", server.Hello)
-	http.HandleFunc("/headers", server.Headers)
-	http.HandleFunc("/calendar", server.Events)
-	http.HandleFunc("/authorisation", server.AuthCode)
-	http.HandleFunc("/countries", server.Countries)
-	
-	http.ListenAndServe(":8080", nil)
+	router := gin.Default()
+	router.Use(cors.Default())
```
Comment (VitaliLupusor) [inline main.go:15]: Default cars allow incoming requests from any origin.

File: cmd/endpoints.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	for _, country := range countries {
+		if country.Name == parameter.Name {
+			context.JSON(200, country)
+			return
+		}
+	}
+
+	message := map[string]string{
+		"error": fmt.Sprintf("Country not found in the database: %s", parameter.Name),
```
Comment (BarryFoye) [inline cmd/endpoints.go:61]: suggest "Country %s is not supported"

File: cmd/helpers.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	if err := json.Unmarshal(data, &countries); err != nil {
+		log.Fatalf("ERROR: Failed to unmarshal the file content: %v", err)
+	}
+
+	return countries
+}
+
+// NOTE: This object is part of V2, where the data is going to be sourced from Google Calendar API
```
Comment (BarryFoye) [inline cmd/helpers.go:49]: then it should be in another branch if not related to this PR

Comment (VitaliLupusor) [reply cmd/helpers.go]: Removed

File: cmd/endpoints.go (hunk 2 above)
Comment (VitaliLupusor) [reply cmd/endpoints.go]: Changed

**Pull request merged**